xbmc/platform/android/activity/XBMCApp.h

```diff
@@ -241,8 +243,9 @@ class CXBMCApp
   bool m_videosurfaceInUse;
   bool m_firstrun;
   bool m_exiting;
+  int m_exitCode{0};
   bool m_bResumePlayback = false;
-  pthread_t m_thread;
+  std::unique_ptr<std::thread> m_thread;
```

Comment:
I don't think you need to use a pointer here.

```suggestion
  std::thread m_thread;
```

Reply:
Right.

Reply:
I need to create the thread in a callback function of CXBMCApp, not at CXBMCApp construction time. std::thread has no copy ctor (understandable). How can I assign the thread worker function to a default constructed std::thread?

Reply:
std::move ? std::thread::swap ? => to "complicated". I do not need two thread instances. I will stick with the unique_ptr approach.

Reply:
`std::thread` default constructor doesn't create a thread.

You should be able to just create it when needed:
```c++
m_thread = std::thread(&CXBMCApp::run, this);

```

Reply:
Okay, thanks. Will change that.